src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/IQueueItem.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+#nullable enable
+
+public interface IQueueItem<RequestContextType>
+{
+    /// <summary>
+    /// Begins executing the work specified by this queue item.
+    /// </summary>
+    Task CallbackAsync(RequestContextType? context, CancellationToken cancellationToken);
+
+    /// <inheritdoc cref="IRequestHandler.RequiresLSPSolution" />
+    bool RequiresLSPSolution { get; }
```

Comment:
Based on [this comment](https://github.com/dotnet/roslyn/pull/62924#discussion_r929444382) I think we should do away with this entirely. Sounds like an impl detail on when a sub-server would want to shutdown. Aka, this is too Roslyny 😄  /cc @dibarbet